add keymap to delete buffer

garymjr opened the pull request:
Adds a new function and builtin keymap to close buffers from the JABS buffer. Also refactored the open function a little bit to cut down on duplication.

matbme commented:
Hi, huge thanks for the help!

I tested your PR on my end and it works mostly fine. I experienced two small bugs I pushed a commit to fix:

- When a buffer is deleted the line numbers appear on JABS' window is `vim.wo.number` or `vim.wo.relativenumber` is set to true. This was fixed by adding

``` lua
vim.wo.number = false
vim.wo.relativenumber = false
```

at the end of the delete function.

- The alternate buffer was showing the symbol and highlight from a hidden buffer. This happened because the `:ls` command was moved to after JABS' window was created, turning the floating window into the alternate buffer when ls is called. I fixed this by making `bopen` a M global and manually removing the deleted line from the table before calling refresh.

Can you pull the commit and check if everything still works on your end?

garymjr commented:
Just pulled it down and it looks good to me! I went ahead and updated the README as well to reflect the new keymap as well as make the existing ones easier to find.

I also updated the existing keymaps you had in the README of `h` and `j` since I _think_ they were a typo and meant to read `j` and `k`. If not I can change those back.

matbme commented:
Great, then merged it is!
